timemachine/ff/handlers/utils.py

```diff
@@ -67,3 +103,224 @@ def match_smirks(mol, smirks):
         matches.append(tuple(mas))
 
     return matches
+
+
+def get_query_mol(mol: Mol) -> Mol:
+    """
+    Return a query mol that has generic bonds for substructure matching.
+    """
+    query_params = Chem.AdjustQueryParameters.NoAdjustments()
+    query_params.makeBondsGeneric = True
+    query_generic_bonds = Chem.AdjustQueryProperties(mol, query_params)
+    return query_generic_bonds
+
+
+def make_residue_mol(name: str, elements: List[str], bonds: List[Tuple[int, int]]) -> Mol:
+    """
+    Generate an rdkit molecule given a list of elements and a list of bonds
+    for a residue.
+
+    Parameters
+    ----------
+    name:
+        Name of the residue.
+    elements:
+        List of atomic elements.
+    bonds:
+        List of atom idxs to connect.
+
+    """
+    mol = Chem.RWMol()
+    mol.BeginBatchEdit()
+    for i, element in enumerate(elements):
+        atom = Chem.Atom(element)
+        atom.SetProp("molAtomMapNumber", str(i))
+        mol.AddAtom(atom)
+
+    for src, dst in bonds:
+        mol.AddBond(src, dst, Chem.BondType.SINGLE)
+
+    mol.CommitBatchEdit()
+    mol = Chem.RemoveHs(mol, implicitOnly=True, sanitize=False)
+    mol.SetProp("_Name", name)
+    return mol
+
+
+def make_residue_mol_from_template(template_name: str) -> Optional[Mol]:
+    """
+    Given a template residue name (e.g. HIP or NALA),
+    return an RDKit molecule that has the proper
+    bond types and charges set. Return None if the residue
+    could not be generated from the template.
+
+    Parameters
+    ----------
+    template_name:
+        Name of the residue. Supported residues are in `SMILES_BY_RES_NAME`.
+
+    Return
+    ------
+        Mol or None if the molecule could not be generated.
+    """
+    # Convert capped to regular residue name
+    res_name, has_n_cap, has_c_cap = get_res_name(template_name)
```

Comment:
[doc imprecision] valid values of `template_name` form a superset of `SMILES_BY_RES_NAME.keys()`

Reply:
updated